tests/common/fixtures/elasticsearch.py

```diff
@@ -18,6 +18,17 @@ def es_client(delete_all_elasticsearch_documents):
     return _es_client()
 
 
+@pytest.fixture
+def es6_client():
+    client = _es6_client()
+    yield client
+    client.conn.delete_by_query(index=client.index, body={"query": {"match_all": {}}},
```

Comment:
I was originally going to write that the rationale for using a parent fixture rather than a `yield` in the `es_client` fixture is because the code after `yield` may not run if eg. the test runner gets killed. However I've realized this is no different than the `delete_all_elasticsearch_documents` fixture which registers a finalizer but doesn't clear the index before the test runs.

One thing I did notice is that the `delete_all_elasticsearch_documents` fixture _is_ also [used by the functional tests](https://github.com/hypothesis/h/blob/ccde7053c9828506e9c8ffceae224daf8d77d028/tests/functional/conftest.py#L81) to ensure a clean state after each functional test is run. We will need to do the same with the ES6 cluster.

Reply:
Related Slack discussion from this morning (UK time): https://hypothes-is.slack.com/archives/C1MA4E9B9/p1532080506000125